src/sparsezoo/model/utils.py

```diff
@@ -562,29 +579,31 @@ def include_file_download_url(files: List[Dict]):
 
 
 def get_model_metadata_from_stub(stub: str) -> Dict[str, str]:
-    """
-    Return a dictionary of the model metadata from stub
-    """
-
-    stub_regex_expr = (
-        r"^(zoo:)?"
-        r"(?P<domain>[\.A-z0-9_]+)"
-        r"/(?P<sub_domain>[\.A-z0-9_]+)"
-        r"/(?P<architecture>[\.A-z0-9_]+)(-(?P<sub_architecture>[\.A-z0-9_]+))?"
-        r"/(?P<framework>[\.A-z0-9_]+)"
-        r"/(?P<repo>[\.A-z0-9_]+)"
-        r"/(?P<dataset>[\.A-z0-9_]+)"
-        r"/(?P<sparse_tag>[\.A-z0-9_-]+)"
-    )
-    matches = re.match(stub_regex_expr, stub)
-
-    return {
-        "domain": matches.group("domain"),
-        "sub_domain": matches.group("sub_domain"),
-        "architecture": matches.group("architecture"),
-        "sub_architecture": matches.group("sub_architecture"),
-        "framework": matches.group("framework"),
-        "repo": matches.group("repo"),
-        "dataset": matches.group("dataset"),
-        "sparse_tag": matches.group("sparse_tag"),
-    }
+    """Return a dictionary of the model metadata from stub"""
+
+    matches = re.match(STUB_V1_REGEX_EXPR, stub) or re.match(STUB_V2_REGEX_EXPR, stub)
+    if matches:
+        if "source_dataset" in matches.groupdict():
+            return {"repo_name": stub}
+
+        if "dataset" in matches.groupdict():
+            return {
+                "domain": matches.group("domain"),
+                "sub_domain": matches.group("sub_domain"),
+                "architecture": matches.group("architecture"),
+                "sub_architecture": matches.group("sub_architecture"),
+                "framework": matches.group("framework"),
+                "repo": matches.group("repo"),
+                "dataset": matches.group("dataset"),
+                "sparse_tag": matches.group("sparse_tag"),
+            }
+
+    return {}
+
+
+def is_stub(candidate: str):
+    if candidate.startswith(ZOO_STUB_PREFIX):
+        return True
+    if bool(get_model_metadata_from_stub(candidate)):
+        return True
+    return False
```

Comment:
I don't know if it matters, but I think I'd handle this `if` statement by directly checking the string instead of seeing if a full metadata dict can be extracted from candidate
```suggestion
def is_stub(candidate: str) -> bool:
    return re.match(STUB_V1_REGEX_EXPR, candidate) or re.match(STUB_V2_REGEX_EXPR, candidate)
```